### ISSUE TITLE
Removing from a repo with mof_compiler script

### ISSUE THREAD

**Issue by KSchopmeyer** [pywbem/pywbem#1158]
Removing items from the repo with mof_compiler appears to be incomplete.  After a correct install,

We executed the following:

```

(pywbem27) kschopmeyer@sheldon:~/pywbem/githubpywbem/pywbem(master)$ mof_compiler -r -s http://sheldon testsuite/test.mof
Error deleting instance root/cimv2:PyWBEM_AllTypes
     6 CIM_ERR_NOT_FOUND: PyWBEM_AllTypes
Error deleting instance root/cimv2:PyWBEM_AllTypes
     6 CIM_ERR_NOT_FOUND: PyWBEM_AllTypes
Error deleting instance root/cimv2:PyWBEM_MemberOfPersonCollection
     6 CIM_ERR_NOT_FOUND: PyWBEM_MemberOfPersonCollection
Error deleting instance root/cimv2:PyWBEM_MemberOfPersonCollection
     6 CIM_ERR_NOT_FOUND: PyWBEM_MemberOfPersonCollection
Error deleting instance root/cimv2:PyWBEM_MemberOfPersonCollection
     6 CIM_ERR_NOT_FOUND: PyWBEM_MemberOfPersonCollection
Error deleting instance root/cimv2:PyWBEM_PersonCollection
     6 CIM_ERR_NOT_FOUND: PyWBEM_PersonCollection
Error deleting class root/cimv2:PyWBEM_AllTypes
     9 CIM_ERR_CLASS_HAS_INSTANCES: PyWBEM_AllTypes
Error deleting class root/cimv2:PyWBEM_MemberOfPersonCollection
     9 CIM_ERR_CLASS_HAS_INSTANCES: PyWBEM_MemberOfPersonCollection
Error deleting class root/cimv2:PyWBEM_PersonCollection
     9 CIM_ERR_CLASS_HAS_INSTANCES: PyWBEM_PersonCollection

```

The results are inconsistent.

1. The PyWBEM_AllTypes class is definitily in the repo.

```

    cimcli gc PyWBEM_AllTypes

    // ===================================================
    // PyWBEM_AllTypes
   // ===================================================

       [Version ( "1.0.0" ), 

```

2. Instances still exist for PyWBEM_MemberOfPersonCollection

```

kschopmeyer@sheldon:~/pgogtest$ cimcli ni PyWBEM_MemberOfPersonCollection
PyWBEM_MemberOfPersonCollection.Collection="PyWBEM_PersonCollection.InstanceID=\"PersonCollection\"",Member="PyWBEM_Person.CreationClassName=\"PyWBEM_Person\",Name=\"Alice\""
PyWBEM_MemberOfPersonCollection.Collection="PyWBEM_PersonCollection.InstanceID=\"PersonCollection\"",Member="PyWBEM_Person.CreationClassName=\"PyWBEM_Person\",Name=\"Bob\""
PyWBEM_MemberOfPersonCollection.Collection="PyWBEM_PersonCollection.InstanceID=\"PersonCollection\"",Member="PyWBEM_Person.CreationClassName=\"PyWBEM_Person\",Name=\"Charlie\""

```

3. Any yet it does delete all of the members of PyWBEM_Person and the class.

```

cimcli ni PyWBEM_Person
cimcli CIMException:  Cmd= ni Object= PyWBEM_Person Code= 5
CIM_ERR_INVALID_CLASS: PyWBEM_Person

```

**Comment by andy-maier** [comment]
**Discussion point:** Mandatory for 0.13.0?

**Comment by andy-maier** [comment]
We decided in the call today that this would be considered optional for 0.13.0

**Comment by andy-maier** [comment]
We decided to move this issue to the next release.

**Comment by andy-maier** [comment]
We decided to put this into 1.0.0 and roll it back to 0.15.0. Setting discussion needed to keep this one above the horizon.

**Comment by KSchopmeyer** [comment]
I know the problem.  It is the compiler again and our use of it in mofcomp -r.  We apparently compile the defined files into the local MOFWBEMConnection so we have a set of names to remove although I have not sorted that out yet exactly.

However, when we send the delete requests for the instances off to the server we do not have a complete path since we compiled them without aliases which is what sets the path component in the compiler.  WHOOPS.  So the server simply cannot find the instances to remove.  Since the server cannot find and remove instances, it also fails any class removal because there are instances.

Here is an example of the request as Pegasus sees it:

```

1573325449s-589284us: XmlIO [3409:139696727299968:HTTPConnection.cpp:2275]: <!-- No request message received; connection closed: queue id: 23 -->
1573325451s-934021us: XmlIO [3409:139696727299968:HTTPConnection.cpp:2319]: <!-- Request: queue id: 23 -->
POST /cimom HTTP/1.1
Host: localhost:5988
Accept-Encoding: identity
Content-type: application/xml; charset="utf-8"
Content-length: 388
PegasusAuthorization: Local "kschopmeyer"
CIMOperation: MethodCall
CIMMethod: DeleteInstance
CIMObject: root/cimv2

<?xml version="1.0" encoding="utf-8" ?>
<CIM CIMVERSION="2.0" DTDVERSION="2.0"><MESSAGE ID="1001" PROTOCOLVERSION="1.0">
<SIMPLEREQ><IMETHODCALL NAME="DeleteInstance"><LOCALNAMESPACEPATH><NAMESPACE NAME="root"/><NAMESPACE NAME="cimv2"/></LOCALNAMESPACEPATH>
<IPARAMVALUE NAME="InstanceName"><INSTANCENAME CLASSNAME="PyWBEM_AllTypes"/></IPARAMVALUE></IMETHODCALL></SIMPLEREQ></MESSAGE></CIM>
1573325451s-936044us: XmlIO [3409:139696727299968:HTTPConnection.cpp:924]: <!-- Response: queue id: 23 -->
HTTP/1.1 200 OK
Content-Type: application/xml; charset=utf-8
content-length: 0000000290
CIMOperation: MethodResponse

<?xml version="1.0" encoding="utf-8" ?>
<CIM CIMVERSION="2.0" DTDVERSION="2.0">
<MESSAGE ID="1001" PROTOCOLVERSION="1.0">
<SIMPLERSP>
<IMETHODRESPONSE NAME="DeleteInstance">
<ERROR CODE="6" DESCRIPTION="CIM_ERR_NOT_FOUND: PyWBEM_AllTypes"/></IMETHODRESPONSE>
</SIMPLERSP>
</MESSAGE>
</CIM>

**Comment by KSchopmeyer** [comment]
Note that pegasus also returns NOT_FOUND for at least one class, Pywbem_Person but I have not sorted out the reason for that.

```
1573325451s-977797us: XmlIO [3409:139696727299968:HTTPConnection.cpp:2275]: <!-- No request message received; connection closed: queue id: 23 -->
1573325451s-979420us: XmlIO [3409:139696727299968:HTTPConnection.cpp:2319]: <!-- Request: queue id: 23 -->
POST /cimom HTTP/1.1
Host: localhost:5988
Accept-Encoding: identity
Content-type: application/xml; charset="utf-8"
Content-length: 372
PegasusAuthorization: Local "kschopmeyer"
CIMOperation: MethodCall
CIMMethod: DeleteClass
CIMObject: root/cimv2

<?xml version="1.0" encoding="utf-8" ?>
<CIM CIMVERSION="2.0" DTDVERSION="2.0"><MESSAGE ID="1001" PROTOCOLVERSION="1.0">
<SIMPLEREQ><IMETHODCALL NAME="DeleteClass"><LOCALNAMESPACEPATH><NAMESPACE NAME="root"/><NAMESPACE NAME="cimv2"/>
</LOCALNAMESPACEPATH><IPARAMVALUE NAME="ClassName">
<CLASSNAME NAME="PyWBEM_Person"/></IPARAMVALUE></IMETHODCALL></SIMPLEREQ></MESSAGE></CIM>
1573325451s-979762us: XmlIO [3409:139696727299968:HTTPConnection.cpp:924]: <!-- Response: queue id: 23 -->
HTTP/1.1 200 OK
Content-Type: application/xml; charset=utf-8
content-length: 0000000285
CIMOperation: MethodResponse

<?xml version="1.0" encoding="utf-8" ?>
<CIM CIMVERSION="2.0" DTDVERSION="2.0">
<MESSAGE ID="1001" PROTOCOLVERSION="1.0">
<SIMPLERSP>
<IMETHODRESPONSE NAME="DeleteClass">
<ERROR CODE="6" DESCRIPTION="CIM_ERR_NOT_FOUND: PyWBEM_Person"/></IMETHODRESPONSE>
</SIMPLERSP>
</MESSAGE>
</CIM>

```

**Comment by KSchopmeyer** [comment]
This is going to take more than a day to fix.

**Comment by andy-maier** [comment]
Will be fixed with PR #1944 (for master) and TBD for 0.15.0.

**Comment by KSchopmeyer** [comment]
**DISCUSSION:** Found two problems (see above) Fixed first but have not figured out why the second is happening.  The question is, whether this is worthwhile for 0.15.0 since it has been broken for many years and nobody noticed. 

I Propose that If I cannot define a fix for the second problem (association being created as part of the remove process with incomplete reference property (keybindings empty) we consider moving this one forward.

**Comment by andy-maier** [comment]
Fixed in PR #1944 for master, and TBD for 0.15.0.
